src/analyzer/protocol/ssl/SSL.cc

```diff
@@ -266,21 +265,21 @@ bool SSL_Analyzer::TryDecryptApplicationData(int len, const u_char* data, bool i
         // server write_key
         const u_char* s_wk = keys.data() + 32;
         // client IV
-        const u_char* c_iv = keys.data() + 64;
+        const char* c_iv = reinterpret_cast<const char*>(keys.data()) + 64;
         // server IV
-        const u_char* s_iv = keys.data() + 68;
+        const char* s_iv = reinterpret_cast<const char*>(keys.data()) + 68;
 
         // FIXME: should we change types here?
-        u_char* encrypted = (u_char*)data;
-        size_t encrypted_len = len;
+        char* encrypted = (char*)data;
+        int encrypted_len = len;
 
         if ( is_orig )
             c_seq++;
         else
             s_seq++;
 
         // AEAD nonce, length 12
-        std::basic_string<unsigned char> s_aead_nonce;
+        std::string s_aead_nonce;
```

Comment:
Switching s_aead_nonce and s_aead_tag to a std::vector<u_char> allows to remove most of the casts here.

https://github.com/zeek/zeek/commit/b04164b78f38eab03b35cf873cced49435e811e1

I'd be okay merging as is, this can be done as a follow-up, too.

Reply:
agreed, I think adding this in is closer than using `std::string`. Added it to the first commit